message.go

```diff
@@ -83,6 +74,37 @@ func (m message) MarshalJSON() ([]byte, error) {
 	return m.json, nil
 }
 
+// setSentAt remarshalls the message with a new sentAt timestamp
+func (m *message) setSentAt(ts time.Time, maxBytes int) (err error) {
+	switch msg := m.msg.(type) {
+	case Alias:
+		msg.SentAt = ts
+		m.msg = msg
+	case Group:
+		msg.SentAt = ts
+		m.msg = msg
+	case Identify:
+		msg.SentAt = ts
+		m.msg = msg
+	case Page:
+		msg.SentAt = ts
+		m.msg = msg
+	case Screen:
+		msg.SentAt = ts
+		m.msg = msg
+	case Track:
+		msg.SentAt = ts
+		m.msg = msg
+	}
```

Comment:
it would be good to add a default case here that errors out with something like

```
	default:
		err = fmt.Errorf("messages with custom types cannot be enqueued: %T", msg)
		return
```

Reply:
[d8cce36](https://github.com/cloudquery/analytics-go/pull/1/commits/d8cce367dbbbb674f1f25257b838c988388dab31)